### DISCUSSION TITLE
Proposed Changes

### DISCUSSION THREAD

**Discussion by rimi** [wikilabs/plugins#58] [Show and tell]
Hi Mario,

please have a look at the following commits in my fork:

-  https://github.com/rimi/pmario-wikilabs-plugins/commit/ac152e8b1d6c375afb63fd96de79ecbdea932a36
i know, there is a performance issue, but for me it is crucial to have that "feature" -> maybe we/you can make it configurable if performance is an issue ?!

- https://github.com/rimi/pmario-wikilabs-plugins/commit/957b54991875c4d1de2ba295cec87adb9c3ddcd7 
i changed it only for the NewChildHere-Logic displayed next to the nodes in the rendered tree (i do not use the other buttons in the tiddler-tools section)

- BTW: https://github.com/rimi/pmario-wikilabs-plugins/commit/04ff5bd6173405f0d347a20b72795ce605da9a0f is still pending in https://github.com/wikilabs/plugins/pulls 

Best wishes,
Mirko

**Comment by pmario** [comment]
Hi, I did test your zip file from 18.Nov. and it did produce duplicated list elements, so I didn't go further. 

A second thing, that may cause the performance hit is `[is[current]]` it is a shortcut for `[all[tiddlers]is[current]]`, which is much slower as `[all[current]]`. 

`[is[current]]` creates a list of all tiddlers and tests each of them, if it is the currentTiddler. You can test this with `[!is[current]]` it will show all tiddlers except the current tiddler. .. so it's slow.

`[all[current]]` ... reads the current tiddler variable. .. done.

**Reply by rimi** [reply]
Hi, yes - the duplication was fixed with https://github.com/rimi/pmario-wikilabs-plugins/commit/dd6b1e35eb1040570393cf749b67ba0aaed600f6.

What about the other proposals in general (besides the filter-speedups)?

Mirko

**Reply by rimi** [reply]
One question though: what about `[all[current]]` vs `[<currentTiddler>]`? If i understand you right, they are the same regarding performance. True?

Mirko

**Reply by pmario** [reply]
yes
